prov/verbs/src/fi_verbs.c

```diff
@@ -907,6 +937,21 @@ fi_ibv_create_ep(const char *node, const char *service,
 	if (ret)
 		return (errno == ENODEV) ? -FI_ENODATA : -errno;
 
+	/* Remove ib_rai entries added by IBACM to prevent wrong
+	 * ib_connect_hdr from being sent in connect request.
+	 * TODO Choose ib_rai if we came here from fi_endpoint */
+	for (rai_current = &_rai; *rai_current;) {
+		struct rdma_addrinfo *rai_next;
+		if ((*rai_current)->ai_family == AF_IB) {
+			rai_next = (*rai_current)->ai_next;
+			(*rai_current)->ai_next = NULL;
+			rdma_freeaddrinfo(*rai_current);
+			*rai_current = rai_next;
+			continue;
```

Comment:
This doesn't update 'rai_previous' to point to the next entry.  It still references rai_current, which can result in a double free when 'rai_previous' is freed.